wheel/src/run_generator.rs

```diff
@@ -166,7 +167,27 @@ pub fn run_block_generator(
     let program =
         unsafe { std::slice::from_raw_parts(program.buf_ptr() as *const u8, program.len_bytes()) };
 
-    match native_run_block_generator(&mut allocator, program, &refs, max_cost, flags) {
+    let r = if (flags & ANALYZE_SPENDS) == 0 {
+        native_run_block_generator(
+            &mut allocator,
+            program,
+            &refs,
+            max_cost,
+            flags,
+            &mut NonePolicy::default(),
```

Comment:
You can switch on just the single different parameter to make this a bit less repetitive.

```suggestion
            if (flags & ANALYZE_SPENDS) == 0 {&mut NonePolicy::default()} else {&mut MempoolPolicy::default()}
```
or whatever the rust ternary operator is.

Reply:
the policy type is a template parameter, so I don't think that would work

Reply:
Originally I was thinking that passing in the policy as an object would open up the possibility to pass in custom state here. But since we don't have a use case for that yet, it would probably be a simplification to just pass in the type, and make the trait require `Default`. The "Policy" needs to be able to keep state, but (for now) there's no need to pass in state.